### DISCUSSION TITLE
Updating number of lanes quest

### DISCUSSION THREAD

**Discussion by NathanARF** [streetcomplete/StreetComplete#5130] [Ideas]
Could it be feasible to update the 'lanes' quest for two-way traffic roads marked as having 3 or more lanes? For example, if a road segment is marked `lanes=3`, the 'lanes' quest could be triggered on this road segment so it can be determined how many lanes there are in each direction (i.e. adding `lanes:forward=*` and `lanes:backward=*`).

**Comment by mnalis** [comment]
Well, if you choose `Each side has a different number of lanes` in _lanes quest_, you can already specify how much lanes each direction has, and it will be tagged exactly as you proposed (with `lanes`, `lanes:forward` and `lanes:backward`). 

Is that what you want to accomplish, or have I misunderstood?

**Comment by NathanARF** [comment]
I am speaking of enabling the lanes quest for roads *already* tagged with *only* `lanes=*` for a value of 3 or higher, so that the number of lanes in each direction can then be inputted by a user solving said quest.

**Reply by mnalis** [reply]
Ah, I see! So it would match on `lanes >= 3 and !lanes:forward and !lanes:backward and oneway!=no`. 

Makes sense to add that information. But probably **only if the number or lanes is odd** -- for even number of lanes, it would be too spammy I think, as vast majority of them are half/half (e.g. 4 lanes is _much_ more likely to be 2+2 than 3+1, 8 lanes is _much_ more likely to be 4+4 than either 6+2 or 7+1 etc).
 But maybe it would makes sense to ask it on  [highway=*_link roads](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dprimary_link)  regardless if number of lanes is odd.

**Comment by westnordost** [comment]
I think it would make sense. Also for invalid values (e.g. `lanes=5`, `lanes:forward=2`, `lanes:backward=2` etc.)

Issue ticket or better yet a PR is welcome.
